expanalysis/results.py

```diff
@@ -1,9 +1,8 @@
-# -*- coding: utf-8 -*-
-"""
-Created on Thu Mar 24 16:03:04 2016
+'''
+analysis/results.py: part of expfactory package
+results class
 
-@author: ian
-"""
+'''
```

Comment:
This looks much better! I think what you want here is more of a validate function - before any operations are done on a result object, you would want to confirm it looks ok. The validate should check for battery, experiments, etc, then when you write any kind of function that uses a result, you can just call the validate function first thing, and it will take care of printing out errors / alerting the user about a misformed result object.